rumqttd/src/link/remote.rs

```diff
@@ -196,16 +202,18 @@ where
         return Err(Error::ZeroKeepAlive);
     }
 
-    // Register this connection with the router. Router replys with ack which if ok will
-    // start the link. Router can sometimes reject the connection (ex max connection limit)
     let empty_client_id = connect.client_id.is_empty();
     let clean_session = connect.clean_session;
 
-    if cfg!(feature = "allow-duplicate-clientid") {
-        if !clean_session && empty_client_id {
-            return Err(Error::InvalidClientId);
-        }
-    } else if empty_client_id {
+    if empty_client_id && !clean_session {
+        let ack = ConnAck {
+            session_present: false,
+            code: ConnectReturnCode::ClientIdentifierNotValid,
+        };
+
+        let packet = Packet::ConnAck(ack, None);
+        network.write(packet).await?;
+
         return Err(Error::InvalidClientId);
     }
```

Comment:
also lets keep the other logic same in the PR

Reply:
which logic?

Reply:
the config flag

Reply:
can you explain the old logic a bit, like why `allow-duplicate-client-id` need that check?

I can't figure out the use of that condition 😅

Reply:
I think the update logic is more or less the same:

previously in that cfg! flag:

if client_Id is empty and clean session is false, we return error ( that is what we do rn as well )

else:
return error if empty client id ( we don't return error as we are generating random client id! as per specs empty client id can't be specified with clean session false, hence we return error in that case. )

so I think this is good to go? wdyt?

Reply:
we can remove that feature completely btw, as this is the only place it is getting used :eyes:

Reply:
yeah make sense